### ISSUE TITLE
[XL] Investigate and map the current fee structure for Parks

### ISSUE THREAD

**Issue by LindsayMacfarlane** [bcgov/parks-data-register#548]
#### Description:

BC Parks fee structure is currently captured in a PDF report. Let's pull that into a reusable data model and data register that can be used and ingested by our and other's products and services. 

#### Acceptance Criteria:

- [x] translate PDF into csv file
- [x] create a data model

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by
- Blocking

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by LindsayMacfarlane** [comment]
Information related to this ticket can be found on the Mural board - https://app.mural.co/t/bcparks2575/m/bcparks2575/1733954637057/213a71a8563963e6445ca71863ddf96448a87f90

**Comment by meredithom** [comment]
Designs updated in [A&R Figma](https://www.figma.com/design/yoMhvO0KZvsM6qoqwuIjpt/Attendance-%26-Revenue?node-id=6066-14908&t=OT40O1EteVI8WDu8-1)

Category list:
[Variance Categories.xlsx](https://github.com/user-attachments/files/18190932/Variance.Categories.xlsx)

Behaviour:
- Dropdown added for variance reason on form pages
- Certain categories have a subcategory that doesn't need to be visible at all times

Notes:
- Decided to omit required state on variance boxes as there is no current way to save a draft. Open to discussions on this. 
- Decided to keep the list the same for each activity as there's not too many options to scroll through.
- Linking to BCParks.ca for EventTypes isn't possible due to the fact that the events often happen at a park level and sub-areas are not synced between the systems.

**Comment by Dianadec** [comment]
Refinement notes:
- some considerations around putting it in our current data register to be aware of
- need to decide on where to store this (which repo? Data register vs. reserve-rec)
- redo name register as data register?
- If this is only about reservations, do a separate one - data model for reserve-rec
- but if others need to access it, needs to be more broad

**Comment by Dianadec** [comment]
Lots of questions coming up related to this work, and questions keep changing. The meeting with the program area is happening today so we'll see what comes out of that.

**Comment by Dianadec** [comment]
@Christopher-walsh22 for this ticket, would it make sense to limit the AC Scope to just the first two bullets, and then we can create a separate ticket for the last 2 bullets (' - put in data register' and ' - create API endpoints with unit tests')
cc @LindsayMacfarlane

**Comment by Christopher-walsh22** [comment]
@Dianadec 

I think that would make sense, I have been pretty hung up with validating and compiling the data. I am still expecting a fair amount of back and fourth with various parks teams so it would probably be good to divide it from information gathering and actually implementing the code.

**Comment by Dianadec** [comment]
Reach out to someone about seniors discounts? - JM to help.

**Comment by Dianadec** [comment]
Todo: split ticket for last 2 bullets

**Comment by Christopher-walsh22** [comment]
I have updated the workbook in teams for review. 

[FeesDoc](https://bcgov.sharepoint.com/:x:/r/teams/031072/Shared%20Documents/%5B2%20-%20Portfolio%5D%20Parks-Rec%20Digital%20Transformation/6%20-%20Portfolio%20Planning/Fee%20register/FeeDoc.xlsx?d=w1f4d20be13ca485e8bb3a7a39704f281&csf=1&web=1&e=9CzTfJ)

I am not sure the appropriate channels for getting this reviewed. But if there is an opportunity to start the process while I work on the data modeling half of this ticket everything should be ready for that. There is a few additional work sheets I have not included because they didn't seem relevant for review.

**Comment by Dianadec** [comment]
@LindsayMacfarlane any suggestions as to which channels we could share this in, or which group of people could review it?

**Comment by LindsayMacfarlane** [comment]
@Dianadec - @Christopher-walsh22 and I chatted about this today. I think a workshop to surface outstanding questions might be helpful at this stage (vs just reaching out through a Teams channel). We started [drafting a potential outline and specific questions](https://app.mural.co/t/bcparks2575/m/bcparks2575/1733954637057/213a71a8563963e6445ca71863ddf96448a87f90?wid=0-1738276533115). Would love your input/ideas.

**Comment by Dianadec** [comment]
@LindsayMacfarlane @Christopher-walsh22 workshop has been scheduled for Feb 12.

**Comment by Dianadec** [comment]
next piece of this work has been started - the data model AC is more of the initial iteration - may change dependent on workshops.

**Comment by Dianadec** [comment]
Next step: Confirm with Jeff B that the excel sheet is good to go then we can close this ticket (send to PO Review)

**Comment by Christopher-walsh22** [comment]
Once we have validation on the data I believe this ticket can be closed. Following the Fee Register Workshop there was a few minor changes to our naming of data but did not effect the structure on a technical level. I have updated the documents in the shared folder for the Fee Register and added a first draft of documentation and instructions on how to use the API with basic query examples.

 [Shared Folder](https://teams.microsoft.com/l/channel/19%3A0de09a184d95440baa6abe4a98081708%40thread.tacv2/%5B2%20-%20Portfolio%5D%20Parks-Rec%20Digital%20Transformation?groupId=cb2c4bec-527d-4097-80b2-6e40e98d36bf)

**Comment by LindsayMacfarlane** [comment]
Capturing feedback for prosperity:

Hi Jeff! I'm hoping you can take another look at the fee register excel sheet and let us know if you have any additional feedback. Chris has taken the feedback from our chat with you and Niamh into consideration. Next steps will be to take this spreadsheet and finalize the data model in the register. Thanks for your continued support!
 
Hi folks, sorry I wasn't able to get to this this week, just wondering if there are any specific components you are looking for feedback on? I should be able to take a look next week.
 
Nothing specific, mostly just curious if you have noticed any errors or have general feedback for the data contained within. My experience with camping fees in has been pretty limited prior to this project, so just looking for an expert opinion on the subject matter 
 
The data has worked great for test data up to this point. So hopefully it seems complete/correct in your opinion  
 
Thanks so much for helping us out Barber, Jeff C ENV:EX!
 
Hi Barber, Jeff C ENV:EX - curious if you've had time to take a look and provide any feedback? 
 
Hi team, sorry I've been sick this week. Just took a look it all looks pretty good to me. One suggestion would be to differentiate drive-in and walk-in frontcountry campsites, as we have many walk-in rates that differ from the regular drive-in rates.
 
Adding a column for "level of service" would be helpful as well as that is how we set prices for different campgrounds
 
as a general question - will there be the ability to add new fee types to this later? e.g. day use fee, boat launch fee, etc
 
I hope you are feeling better! Thanks for this feedback Barber, Jeff C ENV:EX. I'll review with Walsh, Christopher WLRS:EX to see what can be included, but I know he's built the model so that we can be responsive and adaptive to make changes in the future. 
 
I have been thinking about these questions over the morning. And for both of them this would fit the current data model. 
 
For adding new fees, we would be able to add any type of fee assuming that the fees have all of the follow criteria:
 
1) Is found within a protected area. All the fees require an ORCS to be stored in the db
2) Can be found within a park feature - Because there is no definitive list of park features this can be slightly flexible. The supplement used terms only found in one or instances throughout. So if a new type of fees was created say for instance parking lot fee within Rathtrevor campground we would be able create a park feature called "Rathertrevor parking lot" within the db or attach it to the already existing campground. 
3) Is for a specific service - Using your example, boat launch, day use fee would be used here
4) Has a billing method (day, night, week, per use, etc) - We have an extensive list already but can add more to it as needed. 
 
 
Adding the level of service could be done and stored as an option parameter on each fee. So if there was a list we could easily run a script to add this information for each fee as it becomes available. Is this ready now or is this to be considered in the future? 
 
For the Drive-In vs Walk-In are walk in sections of the campgrounds are these typically considered different park features (Upper Loop vs Lower Loop) or are they different activities (Walk in frontcountry camping vs Drive in front country camping). 
 
I could certainly update everything to include this, just curious how you distinguish the two from one another.

**Comment by LindsayMacfarlane** [comment]
Again @Christopher-walsh22 - fantastic work on this is! This wasn't a small task and you took it on with curiosity and enthusiasm. Well done :)
